tests/foreman/ui/test_activationkey.py

```diff
@@ -1388,10 +1390,34 @@ def test_positive_copy_activation_key(self, new_name):
 
         """
 
-        name = gen_string('alpha')
         with Session(self.browser) as session:
-            make_activationkey(
-                session, org=self.org_name, name=name, env='Library')
-            self.assertIsNotNone(self.activationkey.search_key(name))
-            self.activationkey.copy(name, new_name)
+            set_context(session, org=self.org_name)
+            self.navigator.go_to_activation_keys()
+            self.assertIsNotNone(
+                self.activationkey.search_key(self.base_key_name))
+            self.activationkey.copy(self.base_key_name, new_name)
             self.assertIsNotNone(self.activationkey.search_key(new_name))
+
+    @run_only_on('sat')
+    @attr('ui', 'ak', 'implemented')
+    @data(*invalid_names_list())
+    def test_negative_copy_activation_key(self, new_name):
+        """@Test: Create Activation key and fail copying it
+
+        @Feature: Activation key -Negative Copy
+
+        @Steps:
+        1. Create Activation key
+        2. Copy the Activation key with an invalid name
+
+        @Assert: Activation Key copy does not exist
+
+        """
+
```

Comment:
i'll do that for both. The rest of the test file needs some work as well, but i'll leave that for another day.